python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,237 @@
+
+
+class GlueCatalog(Catalog):
+
+    def __init__(self, name: str, properties: Properties):
+        self.client = boto3.client("glue")
+        self.sts_client = boto3.client("sts")
+        super().__init__(name, **properties)
+
+    def _check_response(self, response: Dict[str, Dict[str, str]]):
+        if response['ResponseMetadata']['HTTPStatusCode'] != 200:
+            raise ValueError(f"Got unexpected status code {response['HttpStatusCode']}")
+
+    def _glue_to_iceberg(self, glue_table):
+        # TODO
+        pass
+
+    def _iceberg_to_glue(self, iceberg_table):
+        # TODO
+        pass
+
+    # tested on pre-existing database
+    def create_table(
+            self,
+            identifier: str | Identifier,
+            schema: Schema,
+            location: str | None = None,
+            partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+            sort_order: SortOrder = UNSORTED_SORT_ORDER,
+            properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        now = datetime.now()
+
+        def _convert_column(field: NestedField):
+            d = {'Name': field.name, 'Type': str(field.field_type)}
+
+            if field.doc:
+                d['Comment'] = field.doc
+
+            return d
+
+        # Do all the metadata foo once the Hive PR has been merged
+        try:
+            create_table_response = self.client.create_table(
+                DatabaseName=database_name,
+                TableInput={
+                    'Name': table_name,
+                    'Description': 'string',  # To be fixed
+                    'Owner': properties[OWNER] if properties and OWNER in properties
+                    else boto3.client("sts").get_caller_identity().get("Account"),
+                    'LastAccessTime': now,
+                    'LastAnalyzedTime': now,
+                    'StorageDescriptor': {
+                        'Columns': list(map(_convert_column, schema.fields)),
+                        'Location': location or 's3://',  # To be fixed
+                        'BucketColumns': [
+                            'string',
+                        ],
+                        'SortColumns': [{
+                            schema.find_column_name(field.source_id),
+                            1 if field.direction == SortDirection.ASC else 0
+                        } for field in sort_order.fields]
+                    },
+                    'PartitionKeys': [
+                        {
+                            'Name': schema.find_column_name(spec.source_id),
+                            'Type': str(schema.find_type(spec.source_id)),
+                            'Comment': str(spec.transform)
+                        }
+                        for spec in partition_spec.fields],
+                    'TableType': 'EXTERNAL_TABLE',
+                    'Parameters': properties,
```

Comment:
the code logic here is not quite right. I think we have a misunderstanding of how GlueCatalog works with Iceberg tables. The necessary information stored in Glue is just table name, description, and parameters `table_type=ICEBERG` and `metadata_location=s3://xxxxx`. Everything else are written to S3 as a part of Iceberg base library here https://github.com/apache/iceberg/blob/master/python/pyiceberg/catalog/hive.py#L307-L312. I would suggest read `hive.py` to see how that works and then map to what should be implemented here.

Reply:
Thank you for your suggestions. I'll redesign my implementation following the logic in `hive.py`